openmetadata-service/src/main/java/org/openmetadata/service/search/elasticsearch/ElasticSearchClient.java

```diff
@@ -419,6 +419,79 @@ public Response search(SearchRequest request) throws IOException {
     return Response.status(OK).entity(response).build();
   }
 
+  @Override
+  public Response searchLineage(String fqn, int upstreamDepth, int downstreamDepth, String queryFilter)
+      throws IOException {
+    Map<String, Object> responseMap = new HashMap<>();
+    List<Map<String, Object>> edges = new ArrayList<>();
+    Set<Map<String, Object>> nodes = new HashSet<>();
+    es.org.elasticsearch.action.search.SearchRequest searchRequest =
+        new es.org.elasticsearch.action.search.SearchRequest(GLOBAL_SEARCH_ALIAS);
+    SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+    searchSourceBuilder.query(QueryBuilders.boolQuery().must(QueryBuilders.termQuery("fullyQualifiedName", fqn)));
+    searchRequest.source(searchSourceBuilder);
+    SearchResponse searchResponse = client.search(searchRequest, RequestOptions.DEFAULT);
+    for (var hit : searchResponse.getHits().getHits()) {
+      responseMap.put("entity", hit.getSourceAsMap());
+    }
+    getLineage(fqn, downstreamDepth, edges, nodes, queryFilter, "lineage.fromEntity.fqn.keyword");
+    getLineage(fqn, upstreamDepth, edges, nodes, queryFilter, "lineage.toEntity.fqn.keyword");
+    responseMap.put("edges", edges);
+    responseMap.put("nodes", nodes);
+    return Response.status(OK).entity(responseMap).build();
+  }
+
+  private void getLineage(
+      String fqn,
+      int depth,
+      List<Map<String, Object>> edges,
+      Set<Map<String, Object>> nodes,
+      String queryFilter,
+      String direction)
+      throws IOException {
+    if (depth <= 0) {
+      return;
+    }
+    es.org.elasticsearch.action.search.SearchRequest searchRequest =
```

Comment:
can we move this into import

Reply:
@harshach  We have two SearchRequest one is internal that why we use it this way.